// docker-compose-js/dist/getServiceId.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const executeCommand_1 = require("./executeCommand");
const getDockerComposeCommand_1 = require("./getDockerComposeCommand");
function getServiceId(config) {
    return __awaiter(this, void 0, void 0, function* () {
        const options = {
            env: config.environmentVariables,
            cwd: config.cwd
        };
        const command = getDockerComposeCommand_1.getDockerComposeCommand({
            command: 'ps',
            commandArgs: ['-q', config.serviceName],
            composeFiles: config.composeFiles,
            projectName: config.projectName
        });
        const result = yield executeCommand_1.executeCommand(command, options);
        if (result.error) {
            console.log(result.stderr);
            throw result.error;
        }
        if (!result.stdout) {
            console.log(result.stderr);
            throw new Error('Failed getting the id for the service');
        }
        return result.stdout.replace('\r', '').replace('\n', '');
    });
}
exports.getServiceId = getServiceId;
